blocks/library/list/index.js

```diff
@@ -121,6 +134,31 @@ registerBlockType( 'core/list', {
 					} );
 				},
 			},
+			...[ 'OL', 'UL' ].map( ( tag ) => ( {
+				type: 'shortcut',
+				shortcut: tag.charAt( 0 ).toLowerCase(),
+				transform( attributes ) {
+					const firstNodeName = first( attributes ).nodeName;
+					const isSame = attributes.every( ( { nodeName } ) => nodeName === firstNodeName );
+
+					// All lists already with tag, set back to paragraphs.
+					if ( isSame && firstNodeName === tag ) {
+						return attributes.reduce( ( acc, { values } ) => [
+							...acc,
+							...compact( values.map( ( value ) => get( value, 'props.children', null ) ) )
+								.map( ( content ) => createBlock( 'core/paragraph', {
+									content: [ content ],
+								} ) ),
+						], [] );
+					}
+
+					// Merge list.
+					return createBlock( 'core/list', {
+						nodeName: tag,
+						values: attributes.reduce( ( acc, { values } ) => [ ...acc, ...values ], [] ),
```

Comment:
Maybe a good place for `lodash#flatMap`, per #3185, even though this isn't a hot path like that of the mentioned PR.